drivers/usb/usbccgp/misc.c

```diff
@@ -102,26 +46,14 @@ USBCCGP_SyncUrbRequest(
     IoStack->Parameters.DeviceIoControl.InputBufferLength = UrbRequest->UrbHeader.Length;
     Irp->IoStatus.Status = STATUS_SUCCESS;
 
-    /* Setup completion routine */
-    IoSetCompletionRoutine(Irp,
-                           USBSTOR_SyncForwardIrpCompletionRoutine,
-                           &Event,
-                           TRUE,
-                           TRUE,
-                           TRUE);
-
-    /* Call driver */
-    Status = IoCallDriver(DeviceObject, Irp);
-
-    /* Check if request is pending */
-    if (Status == STATUS_PENDING)
+    if (IoForwardIrpSynchronously(DeviceObject, Irp))
```

Comment:
This is not correct -- making exactly the mistake from https://www.osronline.com/article.cfm%5Earticle=236.htm